File: RepoTransBench/source_projects/Java/cmusphinx_pocketsphinx-android/src/test/java/edu/cmu/pocketsphinx/SpeechRecognizerSetupPublicTest.java
package edu.cmu.pocketsphinx;

import org.junit.Test;
import java.io.File;

import static org.junit.Assert.*;

public class SpeechRecognizerSetupPublicTest {

    @Test
    public void testDefaultSetupReturnsSetupInstancePublic() {
        SpeechRecognizerSetup setup = SpeechRecognizerSetup.defaultSetup();
        assertNotNull(setup);
    }

    @Test
    public void testSetupWithNonExistingFilesPublic() {
        SpeechRecognizerSetup setup = SpeechRecognizerSetup.defaultSetup();
        File acousticModel = new File("dummy_model_dir_public");
        File dictionary = new File("dummy_dict_file_public.dic");
        File logDir = new File("dummy_log_dir_public");

        assertSame(setup, setup.setAcousticModel(acousticModel));
        assertSame(setup, setup.setDictionary(dictionary));
        assertSame(setup, setup.setRawLogDir(logDir));
    }

    @Test
    public void testGetRecognizerReturnsInstancePublic() {
        SpeechRecognizerSetup setup = SpeechRecognizerSetup.defaultSetup();
        assertNotNull(setup.getRecognizer());
    }
}

File: RepoTransBench/target_projects/Java/Go/cmusphinx_pocketsphinx-android/public_tests/speech_recognizer_setup_public_test.go
package public_tests

import (
	"testing"
)

type SpeechRecognizer struct{}

type SpeechRecognizerSetup struct {
	acousticModel, dictionary, rawLogDir string
}

func DefaultSpeechRecognizerSetup() *SpeechRecognizerSetup {
	return &SpeechRecognizerSetup{}
}

func (s *SpeechRecognizerSetup) SetAcousticModel(f string) *SpeechRecognizerSetup {
	s.acousticModel = f
	return s
}
func (s *SpeechRecognizerSetup) SetDictionary(f string) *SpeechRecognizerSetup {
	s.dictionary = f
	return s
}
func (s *SpeechRecognizerSetup) SetRawLogDir(f string) *SpeechRecognizerSetup {
	s.rawLogDir = f
	return s
}
func (s *SpeechRecognizerSetup) GetRecognizer() *SpeechRecognizer {
	return &SpeechRecognizer{}
}

func TestDefaultSetupReturnsSetupInstancePublic(t *testing.T) {
	setup := DefaultSpeechRecognizerSetup()
	if setup == nil {
		t.Fatal("Expected setup to be non-nil")
	}
}

func TestSetupWithNonExistingFilesPublic(t *testing.T) {
	setup := DefaultSpeechRecognizerSetup()
	model := "dummy_model_dir_public"
	dict := "dummy_dict_file_public.dic"
	logDir := "dummy_log_dir_public"
	if setup.SetAcousticModel(model) != setup {
		t.Errorf("SetAcousticModel should return self")
	}
	if setup.SetDictionary(dict) != setup {
		t.Errorf("SetDictionary should return self")
	}
	if setup.SetRawLogDir(logDir) != setup {
		t.Errorf("SetRawLogDir should return self")
	}
}

func TestGetRecognizerReturnsInstancePublic(t *testing.T) {
	setup := DefaultSpeechRecognizerSetup()
	if setup.GetRecognizer() == nil {
		t.Fatal("Expected non-nil SpeechRecognizer from GetRecognizer")
	}
}